anesthetic/plot.py

```diff
@@ -246,6 +251,84 @@ def plot_1d(ax, data, *args, **kwargs):
     return ans
 
 
+def hist_1d(ax, data, *args, **kwargs):
+    """Plot a 1d histogram
+
+    This functions is a wrapper around matplotlib.axes.Axes.hist, or
+    astropy.visualization.hist if it is available. astropy's hist function
+    allows for a more sophisticated calculation of the bins. All remaining
+    keyword arguments are passed onwards.
+
+    Parameters
+    ----------
+    ax: matplotlib.axes.Axes
+        axis object to plot on
+
+    data: numpy.array
+        Uniformly weighted samples to generate kernel density estimator.
+
+    xmin, xmax: float
+        lower/upper prior bound
+        optional, default data.min() and data.max()
+        cannot be None (reverts to default in that case)
+
+    bins: int or list or str (optional)
+        str only works with astropy's hist function. Then it must be one of:
+        - 'blocks' : use bayesian blocks for dynamic bin widths
+        - 'knuth' : use Knuth's rule to determine bins
+        - 'scott' : use Scott's rule to determine bins
+        - 'freedman' : use the Freedman-Diaconis rule to determine bins
+
+    max_bins: int (optional)
+        Only used by astropy.visualization.hist().
+        Maximum number of bins allowed. With more than a few thousand bins
+        the performance of matplotlib will not be great. If the number of
+        bins is large *and* the number of input data points is large then
+        it will take a very long time to compute the histogram.
+
+    Returns
+    -------
+    patches : list or list of lists
+        Silent list of individual patches used to create the histogram
+        or list of such list if multiple input datasets.
+
+    Other Parameters
+    ---------------
+    **kwargs : `~matplotlib.axes.Axes.hist` properties
+
+    """
+    if data.max()-data.min() <= 0:
+        return
+
+    xmin = kwargs.pop('xmin', data.min())
+    xmax = kwargs.pop('xmax', data.max())
```

Comment:
For consistency with the rest of the code, these should probably have ``None`` as defaults

Reply:
changed accordingly